resourcematch-app/src/pages/index/index.page.tsx

```diff
@@ -52,6 +52,7 @@ function Page() {
     })
 
     if (!response.ok) {
+      setError('Request failed')
```

Comment:
Currently this error handling doesn't capture exceptions thrown by the `fetch` itself. To do that, you can wrap things in a `try/catch` like:

```javascript
    try {
      const response = await fetch(`${ENV.RESOURCEMATCH_API}/match`,  {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          model: 'gpt-3.5-turbo-1106',
          books: [slug],
          text: input
        })
      })

      if (!response.ok) {
        throw new Error('Request failed')
      }

      const data = await response.json()
      return data.urls
    } catch(error) {
      setError('Request failed')
    }
```

Note that in :point_up:  the `sertError()` in the check on `response.ok` is removed since both cases will end up at the same code path.